torch/_dynamo/variables/builder.py

```diff
@@ -464,6 +465,9 @@ def index_source(key):
             return AutogradFunctionVariable(
                 value, guards=make_guards(GuardBuilder.FUNCTION_MATCH)
             )
+        elif isinstance(value, torch.autograd.function.FunctionCtx):
+            # The autograd.function context for forward extraction
+            return BlackHoleVariable()
```

Comment:
This won't work if you need to extract `saved_tensors` in backward?

Reply:
Yes, so I wrap it as a ```AutogradFunctionContextVariable``` to solve this problem, the behavior is falling back if users trying to access saved tensors. ```allow_in_graph``` is the workaround way, it skips tracing for the forward/backward functions completely, so it may loss dynamo optimization opportunity.